### DISCUSSION TITLE
Publishing a signed package

### DISCUSSION THREAD

**Discussion by DavidM29** [unoplatform/uno#18631] [Q&A] [answered]
Hello, how can I specify in the msbuild command to sign the app package with a certificate from azure key vault?

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the question. You can review [this documentation](https://learn.microsoft.com/en-us/windows/msix/desktop/sign-with-akv-cert) about this topic.

**Reply by DavidM29** [reply]
Hello @jeromelaban, thanks for the answer, I have followed the documentation but I am still unable to publish the signed package.

For the testing purposes I specified `net9.0-windows` as the only target framework and published the app using the VS GUI. I was able to successfully create the package with the self-signed certificate, however, when I specified the certificate from Azure Key Vault and tried to create the package I got the following error: `A signing key is required in order to package this project. Please specify a PackageCertificateKeyFile or PackageCertificateThumbprint value in the project file.`

To clarify I was able to list and select the certificate I needed for signing.

Could you help with this or should I file this to the visual studio feedback?

**Reply by jeromelaban** [reply]
Thanks for the update.

Anything related to msix tooling and signing is not managed by Uno. The best is to report it through VS feedback or on the WinUI/WinAppSDK repositories.
